### PR TITLE
feat/enterpriseportal: CreateEnterpriseSubscription RPC definition

### PR DESCRIPTION
**Pull request by bobheadxi**
Proposed RPC definition for creating an Enterprise Subscription, which is currently the `createProductSubscription` GraphQL mutation, via Enterprise Portal. The RPC follows patterns suggested for "create" RPCs in https://google.aip.dev/133.

One thing we didn't account for yet is the way AIP suggests handling IDs and names. It raises some interesting points about user-provided ID components in https://google.aip.dev/133#user-specified-ids - I'm pretty sure we don't want to introduce a user-controlled component to the currently randomized subscription ID, _but_ we may want to introduce a "slug" identifier of sorts that must be unique. The default value can use something like https://pkg.go.dev/github.com/gosimple/slug to convert the display name. This would help us ensure duplicate subscriptions for the same customer aren't created.

Closes CORE-149. Implementation is tracked in a follow-up (CORE-155) - for now this will just return "unimplemented".

## Test plan

CI

### CODE REVIEW

File: lib/enterpriseportal/subscriptions/v1/subscriptions.proto [hunk 1]
```diff
@@ -277,6 +282,32 @@ message UpdateEnterpriseSubscriptionResponse {
...
+  // - display_name
+  //
+  // Optional attributes:
+  // - instance_domain
+  EnterpriseSubscription subscription = 2;
+  // A unique identifier for this request. A UUID is recommended.
+  // This request is only idempotent if a `request_id` is provided.
+  //
+  // Learn more: https://google.aip.dev/155.
+  string request_id = 3;
```
Comment (unknwon) [inline lib/enterpriseportal/subscriptions/v1/subscriptions.proto:303]: Feel like a non-necessary part of feature we have to provide now, but if you're feeling like it, go for it!

**Review by unknwon — APPROVED** [review]

File: lib/enterpriseportal/subscriptions/v1/subscriptions.proto (hunk 1 above)
Comment (bobheadxi) [reply lib/enterpriseportal/subscriptions/v1/subscriptions.proto:303]: Fair enough, I'll get rid of it :)

File: lib/enterpriseportal/subscriptions/v1/subscriptions.proto [hunk 2]
```diff
@@ -277,6 +283,27 @@ message UpdateEnterpriseSubscriptionResponse {
   EnterpriseSubscription subscription = 1;
 }
 
+message CreateEnterpriseSubscriptionRequest {
+  // Reserved for potential addition of 'subscription_name' or similar to match
+  // the behaviour described in https://google.aip.dev/133#user-specified-ids.
+  // We may want to introduce a "slug"-type identifier to disambiguate between
+  // subscriptions beyond generated UUIDs.
+  reserved 1;
```
Comment (bobheadxi) [inline lib/enterpriseportal/subscriptions/v1/subscriptions.proto:291]: @unknwon WDYT about this? (also mentioned in PR description)

I woke up this morning and decided maybe we can just ignore AIP on this one 🤔

**Comment by bobheadxi** [comment]
* **#63317** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63317?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63308** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63308?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bobheadxi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/enterpriseportal/subscriptions/v1/subscriptions.proto (hunk 2 above)
Comment (unknwon) [reply lib/enterpriseportal/subscriptions/v1/subscriptions.proto:291]: Hmm, I am not sure when we gonna support this, I feel like if not any time soon, we'd just leave it out. Arguably, instance name is often time a more human-friendly "slug" for subscriptions 😂

Comment (bobheadxi) [reply lib/enterpriseportal/subscriptions/v1/subscriptions.proto:291]: I guess we could require display name be unique?

Comment (bobheadxi) [reply lib/enterpriseportal/subscriptions/v1/subscriptions.proto:291]: Removing the reservation regardless 👍

Comment (unknwon) [reply lib/enterpriseportal/subscriptions/v1/subscriptions.proto]: If both of us are not sure whether we gonna take use of it, let's remove it. I think it comes down to how CE/TA/Sales cares about how they relate a subscription to a customer, that could be customer name, SFDC ID, or something.

**Comment by bobheadxi** [comment]
### Merge activity

* **Jun 18, 3:35 PM EDT**: @bobheadxi merged this pull request with [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63308).

**Pull request merged**